src/awkward/forms/unionform.py

```diff
@@ -220,8 +220,8 @@ def branch_depth(self):
 
     @property
     def fields(self):
-        fieldslists = [cont.fields for cont in self._contents]
-        return list(set.intersection(*[set(x) for x in fieldslists]))
+        all_fields = [f for cont in self._contents for f in cont.fields]
+        return ak._util.unique_list(all_fields)
```

Comment:
I think this is an improvement; `set.intersection` isn't ordered, and we do care about field ordering. This changes makes the order depend upon position in the `contents` list.

Reply:
This was also a mistake :laugh:, I've replaced it with an ordered intersection.